AudioPlayer/src/main/kotlin/com.github.diamondminer88.plugins/AudioPlayer.kt

```diff
@@ -29,246 +29,404 @@ import com.google.android.material.card.MaterialCardView
+                                            globalCurrentPlayer = newPlayer
+                                            globalPlayingUrl = url
+                                            globalIsPlaying = true
+                                            globalIsCompleted = false 
+                                            newPlayer.start()
+                                            restoreUiToGlobalState()
+                                            startTimer()
+                                        }
+                                        newPlayer.setOnCompletionListener {
+                                            globalIsPlaying = false
+                                            globalIsCompleted = true
+                                            restoreUiToGlobalState()
+                                        }
+                                        newPlayer.prepareAsync()
+                                        globalCleanup = {
+                                            try { newPlayer.stop() } catch (_: Exception) {}
+                                            try { newPlayer.release() } catch (_: Exception) {}
+                                            buttonView?.background = playIcon
+                                            sliderView?.progress = 0
+                                            progressView?.text = "0:00 / ${msToTime(duration)}"
+                                            globalCurrentPlayer = null
+                                            globalPlayingUrl = null
+                                            globalIsPlaying = false
+                                            globalIsCompleted = false 
+                                            cancelTimer()
+                                        }
+                                    } catch (e: Exception) {
+                                        Toast.makeText(ctx, "Failed to play audio: ${e.message}", Toast.LENGTH_SHORT).show()
+                                        newPlayer.release()
+                                        restoreUiToGlobalState()
+                                        cancelTimer()
+                                    } finally {
+                                        buttonView?.isEnabled = true
+                                    }
+                                }
+                            }
+                        }
+
+                        restoreUiToGlobalState()
+                        addView(buttonView)
+                        addView(progressView)
+                        addView(sliderView)
+                    }
+
+                    playerCard.addView(playerBar)
+                    card.addView(playerCard)
+                }
+            }
+        }
+
+        patcher.after<StoreMessages>("handleChannelSelected", Long::class.javaPrimitiveType!!) {
+            stopCurrentPlayer()
+            deleteOggCacheFiles(cacheDir = context.cacheDir, oggCacheMap = oggFileCache)
```

Comment:
Don't do this when changing channels, only on start/stop (stop isn't called on app shutdown)